### PR TITLE
test: Add tests for graph compilation resource estimation

### PR DESCRIPTION
**Pull request by max-radin**
## Description

This PR adds some tests that verify that the resource estimations for graph state compilation are correct for some simple cases.

It would be good if the reviewer could take a look at the expected `n_measurement_steps` for `test_get_resource_estimations_for_program_gives_correct_results` and make sure they make sense.

Also @mstechly note that instead of making a test with RZ at some angle besides pi/4, I added instead a test that combines RX and RY. The reason is that I don't know of any rotation angle for which we can get an exact Clifford + T decomposition of RZ (besides the trivial ones that require zero or one T gates).

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [x] I have updated documentation.

### CODE REVIEW

File: tests/benchq/resource_estimation/test_graph_compilation.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            ),
+            {"n_measurement_steps": 4, "n_nodes": 4, "max_graph_degree": 3},
+        ),
+        (
+            QuantumProgram(
+                [Circuit([H(0)] + [CNOT(i, i + 1) for i in range(3)] + [T(1), T(2)])],
+                1,
+                lambda _: [0],
+            ),
+            {"n_measurement_steps": 6, "n_nodes": 6, "max_graph_degree": 5},
```
Comment (mstechly) [inline tests/benchq/resource_estimation/test_graph_compilation.py:43]: I think in this case should have max graph degree equal to 3? If this statement is correct, it should  also has lower `n_measurement_steps` ?

**Review by mstechly — COMMENTED** [review]
One test case looks fishy to me. @AthenaCaesura ?

File: tests/benchq/resource_estimation/test_graph_compilation.py (hunk 1 above)
Comment (mstechly) [reply tests/benchq/resource_estimation/test_graph_compilation.py:43]: I just talked with @AthenaCaesura and it looks ok (90% confidence), but she would need to do some math to be 100%. And also perhaps come with an extra test case.

Comment (AthenaCaesura) [reply tests/benchq/resource_estimation/test_graph_compilation.py:43]: yup! it looksl like you've got the correct max graph degree here! Feel free to merge!

**Review by mstechly — APPROVED** [review]

**Pull request merged**